src/simulator/layout/layout.ts

```diff
@@ -1,29 +1,41 @@
 import { IEdgeBase } from '../../models/edge';
 import { INode, INodeBase, INodePosition } from '../../models/node';
-import { CircleLayout } from './layouts/circle';
+import { CircularLayout, ICircularLayoutOptions } from './layouts/circular';
+import { GridLayout } from './layouts/grid';
+import { HierarchicalLayout, IHierarchicalLayoutOptions } from './layouts/hierarchical';
 
-export enum layouts {
-  DEFAULT = 'default',
-  CIRCLE = 'circle',
+export type LayoutType = 'hierarchical' | 'circular' | 'grid';
+
+export type LayoutSettingsMap = {
+  hierarchical: IHierarchicalLayoutOptions;
+  circular: ICircularLayoutOptions;
+  grid: Record<string, never>;
+};
+
+export interface ILayoutSettings {
+  type: LayoutType;
+  options?: LayoutSettingsMap[LayoutType];
 }
 
 export interface ILayout<N extends INodeBase, E extends IEdgeBase> {
-  getPositions(nodes: INode<N, E>[], width: number, height: number): INodePosition[];
+  getPositions(nodes: INode<N, E>[]): INodePosition[];
 }
 
-export class Layout<N extends INodeBase, E extends IEdgeBase> implements ILayout<N, E> {
-  private readonly _layout: ILayout<N, E> | null;
-
-  private layoutByLayoutName: Record<string, ILayout<N, E> | null> = {
-    [layouts.DEFAULT]: null,
-    [layouts.CIRCLE]: new CircleLayout(),
-  };
-
-  constructor(layoutName: string) {
-    this._layout = this.layoutByLayoutName[layoutName];
-  }
-
-  getPositions(nodes: INode<N, E>[], width: number, height: number): INodePosition[] {
-    return this._layout === null ? [] : this._layout.getPositions(nodes, width, height);
+export class LayoutFactory {
+  static create<N extends INodeBase, E extends IEdgeBase>(
+    width: number,
+    height: number,
+    settings?: Partial<ILayoutSettings>,
+  ): ILayout<N, E> | null {
+    switch (settings?.type) {
+      case 'hierarchical':
+        return new HierarchicalLayout<N, E>(width, height, settings.options as IHierarchicalLayoutOptions);
+      case 'circular':
+        return new CircularLayout<N, E>(width, height, settings.options as ICircularLayoutOptions);
+      case 'grid':
+        return new GridLayout<N, E>(width, height);
+      default:
+        return null;
```

Comment:
Does it make sense to throw an error rather than null? Especially when we will have force layout here.